src/Covenant/Data.hs

```diff
@@ -228,3 +238,63 @@ mkBBF (DataDeclaration _ numVars ctors _)
      - Actually this is slightly false, we need to "bump" all of the indices inside constructor arms by one (because
        they now occur within a Thunk), but after that bump everything is stable as indicated above.
 -}
+
+{- Here for lack of a better place to put it (has to be available to Unification and ASG)
+-}
+
+-- | Packages up all of the relevation datatype information needed
+-- for the ASGBuilder. Note that only certain datatypes have a BB or BB/BF form
+-- (we do not generate forms that are "useless")
+data DatatypeInfo
+  = DatatypeInfo
+  { _originalDecl :: DataDeclaration AbstractTy,
+    _baseFunctor :: Maybe (DataDeclaration AbstractTy),
+    _bbForm :: Maybe (ValT AbstractTy),
```

Comment:
In what cases would we be missing a BB form?

Reply:
`data Void` (and literally no other case, but we allow it so we have to keep open the possibility)